Add treeview articles (#602)
telerik/maui-docs: controls/calendar/migrate-from-xamarin.md

slug: calendar-migrate-from-xamarin
---

# Migrating from Xamarin.Forms Calendar to .NET MAUI Calendar

The Telerik UI for .NET MAUI Calendar control has been designed and built from the ground up as a new control with a new API and significant improvements over its Xamarin counterpart.

> >important The Telerik UI for .NET MAUI Calendar control provides selection - single, multiple and range. The Telerik UI for .NET MAUI Calendar control does not manage appointments like the Telerik Xamarin Calendar does. The appointments will be part of the Telerik .NET MAUI Scheduler control. The Telerik UI for .NET MAUI Scheduler control is planned for R3 2023 release. 

Reviewer: The first and the second sentence aren't logically connected. Why are they part of the same paragraph? You may need to add more context to keep them in the same paragraph. Do you mean that the Xamarin control doesn't support selection?
Instead of the note, consider this structure that is clearer and provides more details. If the information is wrong, change it.

```suggestion
When you plan to migrate from Xamarin to .NET MAUI, consider the following differences in the feature set of the Calendar control:
* Unlike the Xamarin Calendar, the .NET MAUI Calendar allows three selection modes—single, multiple, and range.
* Unlike the Xamarin Calendar, the .NET MAUI Calendar does not manage appointments. In Telerik UI for .NET MAUI, appointments will be handled by the .NET MAUI Scheduler control, which is planned for the R3 2023 release.
```

| Xamarin Calendar | .NET MAUI Calendar |
| ------------- | --------------- |
| `ViewMode`(enum of type `CalendarViewMode`) | `DisplayMode`(enum of type `CalendarDisplayMode`) |

> | Styling the cells, view modes, using custom renderers for styling | New Styling API or the different parts of the calendar and different view modes |

Reviewer: This line is unclear. What is the meaning that you are trying to convey?

> | API for Appointments/Scheduling UI | No Scheduler in Telerik .NET MAUI Calendar - Scheduler will be a separate control |

Reviewer: ```suggestion
| Provides an API for appointments (Scheduling UI) | Does not provide an API for appointments. The scheduling functionality in .NET MAUI will be given by the .NET MAUI Scheduler control, which is planned for R3 2023. |
```

| Xamarin Calendar | .NET MAUI Calendar |
| ------------- | --------------- |
| `ViewMode`(enum of type `CalendarViewMode`) | `DisplayMode`(enum of type `CalendarDisplayMode`) |

> | Styling the cells and the different view modes(DayView, MultiDay, AgendaView) are using custom renderers for styling | New Styling API or the different parts of the calendar and different view modes |

Reviewer: ```suggestion
| Styling the cells and the different view modes (DayView, MultiDay, AgendaView) requires custom renderers. | Styling the distinct parts of the calendar and its view modes is handled by the new Styling API. |
```

> | Provides an API for appointments (Scheduling UI) | Does not provide an API for appointments. The scheduling functionality in .NET MAUI will be given by the .NET MAUI Scheduler control, which is planned for R3 2023. |

Reviewer: ```suggestion
| Provides an API for appointments (Scheduling UI). | Does not provide an API for appointments. The scheduling functionality in .NET MAUI will be given by the .NET MAUI Scheduler control, which is planned for R3 2023. |
```
